jenkins/PerfCI_Grafana_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,168 @@
+pipeline {
+    agent {
+        label 'PerfCi'
+    }
+    environment {
+        CONTACT_EMAIL1 = credentials('perfci_contact_email1')
+        ELASTICSEARCH = credentials('perfci_elasticsearch')
+        ELASTICSEARCH_PORT = credentials('perfci_elasticsearch_port')
+        GRAFANA_URL = credentials('perfci_grafana_url')
+        GRAFANA_API_KEY = credentials('perfci_grafana_api_key')
+        GRAFANA_JSON_PATH = credentials('perfci_grafana_json_path')
+        GRAFANA_FOLDER_NAME = credentials('perfci_grafana_folder_name')
+        MAIN_LIBSONNET_PATH = credentials('perfci_main_libsonnet_path')
+        EMAIL = credentials('perfci_email')
+        USER_NAME = credentials('perfci_user_name')
+        GIT_TOKEN = credentials('perfci_git_token')
+        GIT_REPOSITORY = credentials('perfci_git_repository')
+    }
+
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+                        //sh '[ -d ${WORKSPACE} ] && rm -rf ${WORKSPACE}'
+                    sh '''
+                        if [[ "$(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)" != "" ]]; then
+                            sudo podman rmi -f $(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)
+                        fi
+                    '''
+                }
+            }
+        }
+
+        stage('Checkout') {
+            steps {
+                script {
+                    // Git checkout
+                    git branch: 'main', url: 'https://github.com/redhat-performance/benchmark-runner'
+                }
+            }
+        }
+
+       stage('Install dependencies') {
+            steps {
+                script {
+                    sh 'sudo python3 -m pip install --upgrade pip'
+                    sh 'if [[ -f requirements.txt ]]; then pip install -r requirements.txt; fi'
+                    sh 'pip install --upgrade benchmark-runner'
```

Comment:
You could probably write this as a single command.

Reply:
I prefere to keep each command separately and not mix between it.